### ISSUE TITLE
New feature - Backup

### ISSUE THREAD

**Issue by tnilzon** [hluk/CopyQ#502]
A suggestion for a backup feature.
I think it would be a nice feature to in a simple way be able to make backups of the installed CopyQ. As an example there could be options to make backup of all Tabs, and also another option to make a backup of the CopyQ Settings, and finally an option for a complete backup of both the settings and all tabs in one action. This option should also include restore possibility of previous backups. These backups would be very useful if there is a need for a re-installation of CopyQ, or if you are upgrading the computer operative system and would like to re-install CopyQ in the previous configuration.

/ Tomas Nilzon

**Comment by royharoush** [comment]
+1.
if you need something in the meanwhile, keep in mind that on windows your entire configuration and database files are saved in the config folder in the portable version. so you could backup this folder, or just use the portable version and copy the entire copy folder, if you want to save your config and database.

**Comment by dbkaplun** [comment]
Another +1

**Comment by hluk** [comment]
Well there is already the ancient feature-request #42. I'll try to implement this for new version.

Closing as duplicate of #42.

**Comment by hluk** [comment]
As [I commented](https://github.com/hluk/CopyQ/issues/42#issuecomment-280911320) the new export/import is not suitable for backup. So I'm reopening this.

The easy way is just to back up the whole configuration folder. I don't think this could be in available from GUI. I can write down steps for doing backup (and migrating configuration) in wiki.

Any ideas?

**Comment by hluk** [comment]
[Backup wiki page](https://github.com/hluk/CopyQ/wiki/Backup)

**Comment by unspamme** [comment]
I have a similar difficulty with export & import in Linux Mint 19.3 & would like to be able to backup and restore after the upgrade to version 20.
The GUI export seems like it is saving something but when I go to the import function there is nothing there. Should there be a certain file extension?
The command-line option to import and export data does not seem to work for me, maybe because of my n00bness.
the directory referenced from command line copyq info config seems empty too

**Comment by hluk** [comment]
@unspamme Import expects `*.cpq` files. Hmm, I think exporting should append the suffix automatically.

**Comment by unspamme** [comment]
No, it was not leaving any file that I could find.  I tried specifying an extension of copyq and still could not find in file explorer or copyq's import function.

Well, I did a fresh install of Mint 20 and have not got CopyQ working at all. I am a bit of a n00b and I expected it to work but after installing nothing shows up after launching. I need to do some googling and documentation reading.

**Comment by hluk** [comment]
@unspamme Do you see any warnings or errors in log (F12 shortcut from main window)?

**Comment by hluk** [comment]
Oh, also, if you are using Flatpak version of the app, some directories might be inaccessible.

**Comment by royharoush** [comment]
> 
> 
> As [I commented](https://github.com/hluk/CopyQ/issues/42#issuecomment-280911320) the new export/import is not suitable for backup. So I'm reopening this.
> 
> The easy way is just to back up the whole configuration folder. I don't think this could be in available from GUI. I can write down steps for doing backup (and migrating configuration) in wiki.
> 
> Any ideas?

why is the export\import function not suitable for backup ? I've used it multiple times to on windows (installer version) without any issues.

**Comment by unspamme** [comment]
> Oh, also, if you are using Flatpak version of the app, some directories might be inaccessible.

I am not sure if was flatpak but I think so. I can not open copyq at all now because I upgraded to a fresh install to mint 20 without an export or config files saved.

But now the problem is worse now. Mint 20 has stoped supporting flatpacks and snap installs out of the box. I have been trying to re-enable them things and even got the copyq to show up as installed in software manager, flatpak install from the command line says already installed, and the program shows up in the mint ap menu, but It still will not load

I even got the mint menu to put a startup icon on a panel but nothing happens after I press it, no main window, no status icon, nothing.

**Comment by hluk** [comment]
> why is the export\import function not suitable for backup ? I've used it multiple times to on windows (installer version) without any issues.

@royharoush It depends on what you need to back up. One issue is that it doesn't work well with plugins, like encryption (import/export ignores the encryption keys and backs up encrypted tabs as not encrypted).

Currently, the best solution for complete backup is:
1. Exit CopyQ.
2. Back up the whole configuration directory.

Also, if you're moving configuration to different OS, you need to make sure to rename `copyq.ini` (Windows) to `copyq.conf` (Linux, macOS) and vice versa.

**Comment by Droyk** [comment]
@hluk is there's any way to restore all of one's clipboard history 
There should be an auto backup feature that auto saves the history every day, 
For devs every 30 mins would be a great functionality coz I just lost tons of useful history because of an incorrect command!

I know import/export functions exist but they require manual saving users esp. slugs like hate to do anything manual! so please add an auto backup feature!
